### PR TITLE
[DATA-243] Slave IO, Slave SQL and replication lag checks

### PR DESCRIPTION
**Pull request by juannyG**
KNOWN TODOS:
- No tests yet...coming soon - for now we just need to get the monitors up
- I'll be separating Nagios logic into an interface which this SlaveStatusCheck class will eventually extend.
- Readme - setup, running, examples, etc
- OPEN SOURCE IT!

I'm not doing a merge into master because I want that PR to be cleaner with everything we want the public community to install and consume. This is simply a start to get some monitoring around a current blind spot in production.

For info on Nagios Plugin API requirements: http://nagios.sourceforge.net/docs/3_0/pluginapi.html

@eugene-og @naqeeb @marcosmoyano @evasilchenko 

FYI - @ofirshalom

### CODE REVIEW

File: check_mariadb_slaves.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+            sys.exit(2)
+
+        print "OK - Slave io is running"
+        sys.exit(0)
+
+    def get_slave_status(self, slave_connection):
+        """Run the query!"""
+        try:
+            sql = 'SHOW SLAVE "{0}" STATUS'.format(slave_connection)
+            conn = None
```
Comment (eugene-og) [inline check_mariadb_slaves.py:87]: looks unnecessary

File: check_mariadb_slaves.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        if not lag:
+            print "UNKNOWN - No replication lag reported"
+            sys.exit(3)
+
+        if not self.warning or not self.critical:
+            print "UNKNOWN - Warning and critical thresholds undefined"
+            sys.exit(3)
+
+        lag = int(lag)
+        warning = int(self.warning)
```
Comment (eugene-og) [inline check_mariadb_slaves.py]: if self.warning or self.critical is None, this will raise TypeError

**Comment by eugene-og** [comment]
:+1:

File: check_mariadb_slaves.py (hunk 1 above)
Comment (juannyG) [inline check_mariadb_slaves.py:87]: Helps with the "finally" clause - ensures I'm not checking undeclared variables

**Pull request closed without merging**